File: Cosmos-IOS/Cosmostation/Cosmostation/Model/ExportStarname.swift
//
//  ExportStarname.swift
//  Cosmostation
//
//  Created by 정용주 on 2020/11/12.
//  Copyright © 2020 wannabit. All rights reserved.
//

import Foundation

public struct ExportStarname: Codable {
    var type: String = ""
    var addresses: Array<ExportResource> = Array<ExportResource>()
    
    init() {}
    
    init(_ type: String, _ addresses: Array<ExportResource>) {
        self.type = type
        self.addresses = addresses
    }
    
    public struct ExportResource: Codable {
        var ticker: String = ""
        var address: String = ""
    
        init() {}
        
        init(_ ticker: String, _ address: String) {
            self.ticker = ticker
            self.address = address
        }
    }
    
}

File: app/src/main/java/wannabit/io/cosmostaion/model/ExportStarName.java
package wannabit.io.cosmostaion.model;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class ExportStarName {
    @SerializedName("type")
    public String type;

    @SerializedName("addresses")
    public ArrayList<ExportResource> addresses = new ArrayList<>();

    public ExportStarName() {
    }

    public static class ExportResource {
        @SerializedName("ticker")
        public String ticker;

        @SerializedName("address")
        public String address;

        public ExportResource() {
        }

        public ExportResource(String ticker, String address) {
            this.ticker = ticker;
            this.address = address;
        }
    }
}
